src/ck-core/ckfutures.h

```diff
@@ -72,20 +72,14 @@ namespace ck {
       if (handle_.pe != CkMyPe()) {
         CkAbort("A future's value can only be retrieved on the PE it was created on.");
       }
-      CkMarshallMsg *msg = (CkMarshallMsg *)CkWaitFuture(handle_);
-      PUP::fromMem p(msg->msgBuf);
-      PUP::detail::TemporaryObjectHolder<T> holder;
-      p | holder;
-      return std::move(holder.t);
+      auto msg = static_cast<ck::marshall_msg>(CkWaitFuture(handle_));
+      T result;
+      ck::unmarshall(msg, result);
+      return result;
```

Comment:
This change will prevent the `PUP::reconstruct` constructor from working on type `T` in this circumstance. Would it work to keep `PUP::detail::TemporaryObjectHolder<T> holder;`?